swarm/storage/pusher/protocol.go

```diff
@@ -0,0 +1,71 @@
+package pusher
+
+import (
+	"crypto/rand"
+
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	"github.com/ethereum/go-ethereum/p2p"
+	"github.com/ethereum/go-ethereum/rlp"
+)
+
+const (
+	pssChunkTopic   = "PUSHSYNC_CHUNKS"   // pss topic for chunks
+	pssReceiptTopic = "PUSHSYNC_RECEIPTS" // pss topic for statement of custody receipts
+)
+
+// PubSub is a Postal Service interface needed to send/receive chunks and receipts for push syncing
+type PubSub interface {
+	Register(topic string, handler func(msg []byte, p *p2p.Peer) error) func()
+	Send(to []byte, topic string, msg []byte) error
+	BaseAddr() []byte
+}
+
+// chunkMsg is the message construct to send chunks to their local neighbourhood
+type chunkMsg struct {
+	Addr   []byte // chunk address
+	Data   []byte // chunk data
+	Origin []byte // originator
```

Comment:
Is it a security issue to disclose sender?

Why is it necessary to issue receipts here? Shouldn't the hive configuration of the last node on the route make sure that it's in the neighborhood of the chunk, and thus guarantee that it gets stored? In other words; shouldn't we be able to rely on the pss prox send to work?